File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ele=nums[0], cnt=0;
        for(const int &e: nums) {
            if(e==ele)
                cnt++;
            else {
                if(cnt==0)
                    ele = e, cnt++;
                else
                    cnt--;
            }
        }
        return ele;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int el = nums[0];
        int cnt = 1;
        for (int i = 1; i < nums.length; i++) {
            if (cnt == 0) {
                el = nums[i];
                cnt++;
            } else {
                if (el == nums[i]) {
                    cnt++;
                } else {
                    cnt--;
                }
            }
        }
        return el;
    }
}